pkg/util/shell/shell.go

```diff
@@ -51,3 +55,25 @@ func ExecuteShellCommand(command string, args []string, dir string, stdout io.Wr
 
 	return nil
 }
+
+func DevSpaceExecHandler(ctx context.Context, args []string) error {
+	if len(args) > 0 {
+		hc := interp.HandlerCtx(ctx)
+		_, err := interp.LookPathDir(hc.Dir, hc.Env, args[0])
```

Comment:
Are you sure your cat implementation is ever executed in the shell unit tests currently? It would be probably a good idea to force execution of your cat implementation by overriding the `interp.LookPathDir` function like this:
```
var lookPathDir = interp.LookPathDir

func DevSpaceExecHandler(ctx context.Context, args []string) error {
	if len(args) > 0 {
		hc := interp.HandlerCtx(ctx)
		_, err := lookPathDir(hc.Dir, hc.Env, args[0])
```

and then during the unit tests you override the function with one that always returns an error.

Reply:
Done. PTAL